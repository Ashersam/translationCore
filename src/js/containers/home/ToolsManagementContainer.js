import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import ToolsCards from "../../components/home/toolsManagement/ToolsCards";
import HomeContainerContentWrapper
  from "../../components/home/HomeContainerContentWrapper";
import * as ProjectDetailsActions from "../../actions/ProjectDetailsActions";
import * as ProjectDetailsHelpers from "../../helpers/ProjectDetailsHelpers";
import {
  getTools, getIsUserLoggedIn, getProjectSaveLocation, getProjectBookId
} from "../../selectors";
import { openTool } from "../../actions/ToolActions";
import { openAlertDialog } from "../../actions/AlertModalActions";

class ToolsManagementContainer extends Component {
  constructor(props) {
    super(props);
    this.handleSelectTool = this.handleSelectTool.bind(this);
  }

  componentDidMount() {
    const {tools, reducers} = this.props;
    const projectSaveLocation = getProjectSaveLocation(reducers);
    const bookId = getProjectBookId(reducers);
    if (projectSaveLocation && bookId) {
      tools.forEach(({name:toolName}) => {
        this.props.actions.loadCurrentCheckCategories(toolName, bookId, projectSaveLocation);
      });
    }
  }

  /**
   *
   * @param toolName
   */
  handleSelectTool(toolName) {
    const {isUserLoggedIn, openTool, translate, openAlertDialog} = this.props;
    if(isUserLoggedIn) {
      openTool(toolName);
    } else {
      openAlertDialog(translate("please_log_in"));
    }
  }

  render() {
    const {
      tools,
      reducers: {
        loginReducer: { loggedInUser },
        projectDetailsReducer: {
          manifest,
          projectSaveLocation,
          currentProjectToolsProgress,
          currentProjectToolsSelectedGL,
          toolsCategories
        },
        invalidatedReducer
      },
      translate
    } = this.props;
    const instructions = (
      <div>
        <p>{translate("tools.select_tool_from_list")}</p>
        <p>{translate("projects.books_available",
          { app: translate("_.app_name") })}</p>
      </div>
    );
    const availableCategories = ProjectDetailsHelpers.getAvailableCheckCategories(currentProjectToolsSelectedGL);
    return (
      <HomeContainerContentWrapper
        translate={translate}
        instructions={instructions}
      >
        <div style={{ height: "100%" }}>
          {translate("tools.tools")}
          <ToolsCards
            tools={tools}
            onSelectTool={this.handleSelectTool}
            availableCategories={availableCategories}
            toolsCategories={toolsCategories}
            manifest={manifest}
            translate={translate}
            bookName={name}
            loggedInUser={loggedInUser}
            actions={{
              ...this.props.actions
            }}
            invalidatedReducer={invalidatedReducer}
            projectSaveLocation={projectSaveLocation}
            currentProjectToolsProgress={currentProjectToolsProgress}
          />
        </div>
      </HomeContainerContentWrapper>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isUserLoggedIn: getIsUserLoggedIn(state),
    tools: getTools(state),
    reducers: {
      homeScreenReducer: state.homeScreenReducer,
      projectDetailsReducer: state.projectDetailsReducer,
      loginReducer: state.loginReducer,
      invalidatedReducer: state.invalidatedReducer
    }
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    openTool: name => dispatch(openTool(name)),
    openAlertDialog: message => dispatch(openAlertDialog(message)),
    actions: {
      loadCurrentCheckCategories: (toolName, bookName, projectSaveLocation) => {
        dispatch(ProjectDetailsActions.loadCurrentCheckCategories(toolName, bookName, projectSaveLocation));
      },
      getProjectProgressForTools: (toolName) => {
        dispatch(ProjectDetailsActions.getProjectProgressForTools(toolName));
      },
      setProjectToolGL: (toolName, selectedGL) => {
        dispatch(ProjectDetailsActions.setProjectToolGL(toolName, selectedGL));
      },
      updateCheckSelection: (id, value, toolName) => {
        dispatch(
          ProjectDetailsActions.updateCheckSelection(id, value, toolName));
      }
    }
  };
};

ToolsManagementContainer.propTypes = {
  isUserLoggedIn: PropTypes.bool.isRequired,
  openTool: PropTypes.func.isRequired,
  openAlertDialog: PropTypes.func.isRequired,
  tools: PropTypes.array.isRequired,
  reducers: PropTypes.shape({
    projectDetailsReducer: PropTypes.object.isRequired,
    loginReducer: PropTypes.shape({
      loggedInUser: PropTypes.bool
    }).isRequired
  }).isRequired,
  actions: PropTypes.object.isRequired,
  translate: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ToolsManagementContainer);
