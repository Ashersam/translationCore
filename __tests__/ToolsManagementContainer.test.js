/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import ToolsManagementContainer from '../src/js/containers/home/ToolsManagementContainer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('../src/js/components/home/toolsManagement/ToolsCards', () => 'ToolsCards');

describe('ToolsManagementCotnainer container tests', () => {

  test('Check ToolsManagementContainer container', () => {
    const props = {
      reducers: {
        toolsReducer: {
          toolsMetadata: {}
        },
        loginReducer: {
          loggedInUser: {}
        },
        settingsReducer: {
          currentSettings: {
            developerMode: false
          }
        },
        projectDetailsReducer: {
          manifest: {},
          projectSaveLocation: 'save/location',
          currentProjectToolsProgress: {},
          currentProjectToolsSelectedGL: {}
        }
      },
      translate: k=>k
    };
    const wrapper = shallow(<ToolsManagementContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    
    expect(renderedValue).toMatchSnapshot();
  });
});
