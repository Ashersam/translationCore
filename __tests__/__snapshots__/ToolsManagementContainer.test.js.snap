// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolsManagementCotnainer container tests Check ToolsManagementContainer container instructions 1`] = `
<HomeContainerContentWrapper
  instructions={
    <div>
      <p>
        home.tools.select_from_list
      </p>
      <p>
        home.project.supported_projects
      </p>
    </div>
  }
  translate={[Function]}
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  >
    Tools
    <ToolsCards
      actions={
        Object {
          "getProjectProgressForTools": [Function],
          "getToolsMetadatas": [Function],
          "launchTool": [Function],
          "setProjectToolGL": [Function],
        }
      }
      bookName="nodejs"
      currentProjectToolsProgress={Object {}}
      currentProjectToolsSelectedGL={Object {}}
      developerMode={false}
      loggedInUser={true}
      manifest={Object {}}
      projectSaveLocation="save/location"
      toolsMetadata={Array []}
      translate={[Function]}
    />
  </div>
</HomeContainerContentWrapper>
`;
